### PR TITLE
Phone Do not Disturb

### PR DESCRIPTION
**Pull request by realforest2001**
# About the pull request
Allows phones to be put on Do Not Disturb, preventing them from being called. This is not possible to toggle on certain phones such as Overwatch.

# Explain why it's good for the game
Sometimes phones get spammed for trolls, other times the relevant people (like CL) are not in their office. This makes it easy to stop wasted calls on empty rooms or busy people.
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
add: Added a Do Not Disturb feature for MOST phones. (Overwatch and a couple others excluded).
/:cl:

### CODE REVIEW

File: code/modules/cm_phone/phone.dm [hunk 1]
```diff
@@ -24,7 +24,7 @@ GLOBAL_LIST_EMPTY_TYPED(transmitters, /obj/structure/transmitter)
 	var/range = 7
 
 	var/enabled = TRUE
-	var/callable = TRUE
+	var/do_not_disturb = PHONE_DND_OFF
```
Comment (BeagleGaming1) [inline code/modules/cm_phone/phone.dm:27]: code doc

**Review by BeagleGaming1 — CHANGES_REQUESTED** [review]

**Review by BeagleGaming1 — APPROVED** [review]
code good

File: maps/map_files/USS_Almayer/USS_Almayer.dmm [hunk 2]
```diff
@@ -7951,6 +7951,7 @@
 /obj/structure/transmitter/rotary{
 	name = "Bravo Overwatch Telephone";
 	phone_category = "Command";
+	do_not_disturb = -1;
```
Comment (morrowwolf) [inline maps/map_files/USS_Almayer/USS_Almayer.dmm:7954]: Make a subtype with it being set to -1

**Review by morrowwolf — CHANGES_REQUESTED** [review]

**Comment by Cthulhu80** [comment]
Currently, when the button is pressed it doesn't change its color or update the button text or have any visible state change in tgui, it just gives you a little message in the chatbox. It feels like it should though...

**Comment by realforest2001** [comment]
> Currently, when the button is pressed it doesn't change its color or update the button text or have any visible state change in tgui, It feels like it should though...

The hover-text and the icon on the button do update when the state is changed

**Comment by Cthulhu80** [comment]
I didn't notice it in-game when I first commented. I see it in the code.

**Review by Zonespace27 — APPROVED** [review]
I dunno why this has sat so long

**Pull request merged**